src/components/appSwitch/AppSwitch.js

```diff
@@ -1,149 +1,133 @@
+    AsyncStorage.removeItem('destinationPath')
+    log.debug('getParams', { destinationPath, router, state })
+    if (destinationPath) {
+      const app = router.getActionForPathAndParams(destinationPath.path)
       const destRoute = actions => (some(actions, 'action') ? destRoute(actions.action) : actions.action)
-      const destData = { ...destRoute(app), params: navInfo.params }
+      const destData = { ...destRoute(app), params: destinationPath.params }
       return destData
-    } else if (destinationPath) {
-      return JSON.parse(destinationPath)
     }
     return undefined
   }
 
-  //TODO: add shouldComponentUpdate to rerender only on route change/dialog?
-  async componentDidUpdate() {
+  /*
+  If a user has a saved destination path from before logging in or from inside-app (receipt view?)
+  He won't be redirected in checkAuthStatus since it is called on didmount effect and won't happen after
+  user completes signup and becomes loggedin which just updates this component
+*/
+  const navigateToUrlAction = async () => {
     log.info('didUpdate')
-    const destinationPath = await AsyncStorage.getItem('destinationPath')
+    let destDetails = await getParams()
 
     //once user logs in we can redirect him to saved destinationpath
-    if (destinationPath && this.props.store.get('isLoggedIn')) {
-      const destDetails = JSON.parse(destinationPath)
-      await AsyncStorage.removeItem('destinationPath')
+    if (destDetails) {
       log.debug('destinationPath found:', destDetails)
-      return this.props.navigation.navigate(destDetails)
+      return props.navigation.navigate(destDetails)
     }
   }
 
   /**
    * Check's users' current auth status
    * @returns {Promise<void>}
    */
-  checkAuthStatus = async () => {
-    const { credsOrError, isLoggedInCitizen, isLoggedIn } = await Promise.all([
-      checkAuthStatus(this.props.store),
-      delay(TIMEOUT)
-    ]).then(([authResult]) => authResult)
-    let destDetails = await this.getParams()
-    if (isLoggedIn) {
-      if (isLoggedInCitizen) {
-        API.verifyTopWallet()
-      }
-
-      if (destDetails) {
-        this.props.navigation.navigate(destDetails)
-        return AsyncStorage.removeItem('destinationPath')
-      }
-      this.props.navigation.navigate('AppNavigation')
-    } else {
-      const { jwt } = credsOrError
-      if (jwt) {
-        log.debug('New account, not verified, or did not finish signup', jwt)
+  const initialize = async () => {
+    //after dynamic routes update, if user arrived here, then he is already loggedin
+    //initialize the citizen status and wallet status
+    const { isLoggedInCitizen, isLoggedIn } = await Promise.all([getLoginState(), updateWalletStatus(gdstore)]).then(
+      ([authResult, _]) => authResult
+    )
+    gdstore.set('isLoggedIn')(isLoggedIn)
+    gdstore.set('isLoggedInCitizen')(isLoggedInCitizen)
+    isLoggedInCitizen ? API.verifyTopWallet() : Promise.resolve()
```

Comment:
I think you're missing a return here.

Reply:
you mean for verifytopwallet? I dont think so. I dont want to await for the top wallet to finish